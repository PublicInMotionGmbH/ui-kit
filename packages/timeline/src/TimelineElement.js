import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

const moduleName = 'timeline-element'

const propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Content to put inside */
  children: PropTypes.node,

  /** Time to render in component */
  time: PropTypes.string
}

/**
 * Component which represents timeline extended elements.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {node} [props.children]
 * @param {string} [props.time]
 * @returns {React.Element}
 */
function TimelineElement (props) {
  const { time, children, className } = props

  const clsName = buildClassName(moduleName, className)
  const clsNameIcon = buildClassName([moduleName, 'icon'])
  const clsNamePointer = buildClassName([moduleName, 'pointer'])
  const clsNameTime = buildClassName([moduleName, 'time'])

  const timeElement = time ? (
    <div className={clsNameTime}>
      {time}
    </div>
  ) : null

  return (
    <div className={clsName}>
      <div className={clsNameIcon}>
        {children}
      </div>
      <div className={clsNamePointer} />
      {timeElement}
    </div>
  )
}

TimelineElement.displayName = 'TimelineElement'

TimelineElement.propTypes = propTypes

export default TimelineElement
