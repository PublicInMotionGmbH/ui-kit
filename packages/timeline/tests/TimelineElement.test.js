import React from 'react'
import { shallow } from 'enzyme'

import TimelineElement from '../src/TimelineElement'

const moduleName = 'talixo-timeline-element'

describe('<TimelineLine />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(<TimelineElement />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders class name correctly', () => {
    const wrapper = shallow(<TimelineElement />)

    expect(wrapper.prop('className')).toBe(moduleName)
  })

  it('renders time corectly', () => {
    const wrapper = shallow(<TimelineElement time='13:45' />)

    expect(wrapper.find(`.${moduleName}__time`)).toHaveLength(1)
  })

  it('doesn`t renders time when not passed', () => {
    const wrapper = shallow(<TimelineElement />)

    expect(wrapper.find(`.${moduleName}__time`)).toHaveLength(0)
  })
})
