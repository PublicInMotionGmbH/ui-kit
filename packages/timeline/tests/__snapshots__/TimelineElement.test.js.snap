// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineLine /> renders children correctly 1`] = `
<div
  className="talixo-timeline-element"
>
  <div
    className="talixo-timeline-element__icon"
  />
  <div
    className="talixo-timeline-element__pointer"
  />
</div>
`;
